File: baekjoon/6378.cpp
#include <iostream>
using namespace std;

int digital_root(int n)
{
    int ret = 0;
    while (n) {
        ret += n % 10;
        n /= 10;
    }
    if (ret < 10) return ret;
    return digital_root(ret);
}
int main()
{
    int n;
    while (1) {
        cin >> n;
        if (n == 0) break;
        cout << digital_root(n) << '\n';
    }
    return 0;
}

File: baekjoon/6000/6378.py
def digital_root(n):
    ret = 0
    while n > 0:
        ret += n % 10
        n //= 10
    if ret < 10:
        return ret
    return digital_root(ret)


while True:
    N = int(input())
    if N == 0:
        break
    print(digital_root(N))
